python/ray/util/sgd/v2/trainer.py

```diff
@@ -444,6 +409,66 @@ def to_tune_trainable(self, train_func: Callable[[Dict[str, Any]], T]
                                       self._num_workers, self._use_gpu,
                                       self._resources_per_worker)
 
+    def to_workers(self, train_cls: Type, *args, **kwargs) -> "Workers":
+        """Returns Ray actors with the provided class and the backend started.
+
+        This is useful if you want to provide your own class for training
+        and have more control over execution, but still want to use Ray SGD
+        to setup the appropriate backend configurations (torch, tf, etc.).
+
+        .. code-block:: python
+
+            class Trainer:
+                def __init__(self, config):
+                    self.config = config
+
+                def train_epoch(self):
+                    ...
+                    return 1
+
+            config = {"lr": 0.1}
+            trainer = Trainer(num_workers=2, backend="torch")
+            workers = trainer.to_workers(train_cls=Trainer, config=config)
+            futures = [w.train_epoch.remote() for w in workers]
+            assert ray.get(futures) == [1, 1]
+            assert ray.get(workers[0].train_epoch.remote()) == 1
+            workers.shutdown()
+
+        Args:
+            train_cls (Type): The class definition to use for the Ray
+                actors/workers.
+            *args, **kwargs: Arguments to pass into the __init__ of the
+                provided ``train_cls``.
+        """
+        if self._executor.is_started:
+            raise RuntimeError("The Trainer must not be active to use "
+                               "`to_workers`. Either shutdown the "
+                               "Trainer or don't start it in the first place.")
+        self._executor.start(
+            train_cls=train_cls, train_cls_args=args, train_cls_kwargs=kwargs)
+        return Workers(self._executor.worker_group)
+
+
+class Workers:
```

Comment:
Can you add a docstring for this?

This is a public-facing object

Reply:
nit: Plural class name seems weird to me... what are your thoughts on naming it something like `SGDWorkerGroup`?

Reply:
Good idea, changed this to `SGDWorkerGroup`.

Reply:
Also added a docstring.